internal/lookoutingesterv2/instructions/instructions.go

```diff
@@ -0,0 +1,553 @@
+		case *armadaevents.EventSequence_Event_JobRunErrors:
+			err = c.handleJobRunErrors(ts, event.GetJobRunErrors(), update)
+		case *armadaevents.EventSequence_Event_JobDuplicateDetected:
+			err = c.handleJobDuplicateDetected(ts, event.GetJobDuplicateDetected(), update)
+		case *armadaevents.EventSequence_Event_CancelJob:
+		case *armadaevents.EventSequence_Event_JobRunLeased:
+		case *armadaevents.EventSequence_Event_ReprioritiseJobSet:
+		case *armadaevents.EventSequence_Event_CancelJobSet:
+		case *armadaevents.EventSequence_Event_ResourceUtilisation:
+		case *armadaevents.EventSequence_Event_StandaloneIngressInfo:
+		case *armadaevents.EventSequence_Event_JobRunPreempted:
+			log.Debugf("Ignoring event type %T", event)
+		default:
+			log.Warnf("Ignoring unknown event type %T", event)
+		}
+		if err != nil {
+			c.metrics.RecordPulsarMessageError(metrics.PulsarMessageErrorProcessing)
+			log.WithError(err).Warnf("Could not convert event at index %d.", idx)
+		}
+	}
+}
+
+func (c *InstructionConverter) handleSubmitJob(
+	queue string,
+	owner string,
+	jobSet string,
+	ts time.Time,
+	event *armadaevents.SubmitJob,
+	update *model.InstructionSet,
+) error {
+	jobId, err := armadaevents.UlidStringFromProtoUuid(event.GetJobId())
+	if err != nil {
+		c.metrics.RecordPulsarMessageError(metrics.PulsarMessageErrorProcessing)
+		return err
+	}
+
+	// Try and marshall the job Json. This shouldn't go wrong but if it does, it'c not a fatal error
+	// Rather it means that the json won't be available in the ui
+	var jobProto []byte
+	apiJob, err := eventutil.ApiJobFromLogSubmitJob(owner, []string{}, queue, jobSet, ts, event)
+	if err == nil {
+
+		jobProtoUncompressed, err := proto.Marshal(apiJob)
+		if err != nil {
+			log.Warnf("Couldn't marshall job %s in jobset %s as json.  %+v", jobId, jobSet, err)
+		}
+
+		jobProto, err = c.compressor.Compress(jobProtoUncompressed)
+		if err != nil {
+			log.Warnf("Couldn't compress proto for job %s in jobset %s as json.  %+v", jobId, jobSet, err)
+		}
+	} else {
+		c.metrics.RecordPulsarMessageError(metrics.PulsarMessageErrorProcessing)
+		log.Warnf("Couldn't convert job event for job %s in jobset %s to api job.  %+v", jobId, jobSet, err)
+	}
+
+	resources := getJobResources(apiJob)
+	priorityClass := getJobPriorityClass(apiJob)
+
+	job := model.CreateJobInstruction{
+		JobId:                     jobId,
+		Queue:                     queue,
+		Owner:                     owner,
+		JobSet:                    jobSet,
+		Cpu:                       resources.Cpu,
+		Memory:                    resources.Memory,
+		EphemeralStorage:          resources.EphemeralStorage,
+		Gpu:                       resources.Gpu,
+		Priority:                  int64(event.Priority),
+		Submitted:                 ts,
+		LastTransitionTime:        ts,
+		LastTransitionTimeSeconds: ts.Unix(),
+		State:                     database.JobQueuedOrdinal,
+		JobProto:                  jobProto,
+		PriorityClass:             priorityClass,
```

Comment:
do we need to trim any of these string fields to ensure they fit in the varchar cols?

Reply:
this goes for all the string values in this file eg. the jobset in user annotation lookup. Apologies if you are trimming them- I haven't got this in an ide so it's not the easiest thing to check.

Reply:
Truncating all fields (except for job id/run id, as those are passed in as UUIDs)